backend/src/Edna/Config/Environment.hs

```diff
@@ -0,0 +1,97 @@
+module Edna.Config.Environment
+  ( apiListenEnv
+  , askDebugDB
+  , connectionEnv
+  , dbMaxConnectionsEnv
+  , loggingLevelEnv
+  , serveDocsEnv
+  ) where
+
+import Universum
+
+import Control.Exception (throwIO)
+import Data.ByteString.Char8 (pack)
+import qualified Data.Char as C
+import Edna.Config.Definition (LoggingConfig(..), parseLoggingConfig)
+import Edna.Util (ConnString(..), NetworkAddress)
+import System.Environment (lookupEnv)
+import qualified Text.Show
+
+data EnvParseError = WrongNetworkAddress
+                   | WrongMaxConnections
+                   | WrongLoggingLevel
+
+instance Show EnvParseError where
+  show WrongNetworkAddress = "Can't parse EDNA_API_LISTEN"
+  show WrongMaxConnections = "Can't parse EDNA_DB_MAX_CONNECTIONS"
+  show WrongLoggingLevel   = "Can't parse EDNA_LOGGING_LEVEL"
+
+instance Exception EnvParseError
+
+parseEnv :: (EnvParseError -> String -> Either EnvParseError a)
+         -> String
+         -> EnvParseError
+         -> IO (Maybe a)
+parseEnv parser name err = do
+  env <- lookupEnv name
+  case parser err <$> env of
+    Nothing -> return Nothing
+    Just (Right value) -> return $ Just value
+    Just (Left e) -> throwIO e
+
+readEnv :: Read a => String -> EnvParseError -> IO (Maybe a)
+readEnv = parseEnv read
+  where
+    read err name = case readEither name of
+      Left _ -> Left err
+      Right x -> Right x
+
+getBoolEnv :: String -> IO (Maybe Bool)
+getBoolEnv name = lookupEnv name <&> \case
+  Nothing                        -> Nothing
+  Just "1"                       -> Just True
+  Just (map C.toLower -> "true") -> Just True
+  _                              -> Just False
+
+-- If @EDNA_DB_DEBUG@ environment variable is set to @1@
+-- or @TRUE@ (case-insensitive), we will run postgres with
+-- debug logging.
+{-# ANN askDebugDB ("HLint: ignore Use Just" :: Text) #-}
+askDebugDB :: IO Bool
+askDebugDB = do
+  env <- getBoolEnv "EDNA_DB_DEBUG"
+  return $ fromMaybe False env
+
+apiListenEnv :: IO (Maybe NetworkAddress)
+apiListenEnv = readEnv "EDNA_API_LISTEN" WrongNetworkAddress
+
+serveDocsEnv :: IO (Maybe Bool)
+serveDocsEnv = getBoolEnv "EDNA_SERVE_DOCS"
+
+buildConnectionOption :: ByteString -> Maybe String -> Maybe ByteString
+buildConnectionOption option env = (<>) <$> Just option <*> (pack <$> env)
```

Comment:
1. I think it can be simplified as follows: `(option <>) <$> (pack <$> env)` or even `(option <>) . pack <$> env`.
2. `Data.ByteString.Char8.pack` is dangerous because it truncates all characters to 8 bits, it's very rarely a good idea to use it. I think you need `encodeUtf8` here.

Reply:
thanks
fixed

Reply:
And can you please test whether it works if you use special characters (e. g. Russian letters) in password for example? I think the version with `pack` shouldn't work and the version with `encodeUtf8` should work.

Reply:
With `POSTGRES_PASSWORD="тест"` I have `password=B5AB` in connection string in case of `pack` version, and the same one with `encodeUtf8` with `--dump-config`
But in repl:
```
Data.ByteString.Char8 Universum
λ> (option <>) . encodeUtf8 <$> env
Just "test\209\130\208\181\209\129\209\130"
it :: Maybe ByteString
(0.02 secs, 1,376,560 bytes)

Data.ByteString.Char8 Universum
λ> (option <>) . pack  <$> env
Just "testB5AB"
it :: Maybe ByteString
(0.01 secs, 1,353,136 bytes)
```

So I the problem is somewhere in dump-config

Reply:
With `POSTGRES_PASSWORD="тест"` I have `password=B5AB` in connection string in case of `pack` version, and `password= тест` in case of `encodeUtf8` (with `--dump-config`)

Reply:
> (with `--dump-config`)

Okay, and does it actually connect to DB if you run `docker-compose up postgres` and `edna-server` with `POSTGRES_PASSWORD="тест"`? I. e. did you test it without `--dump-config`? Just in case.

Reply:
it's working with local postgres and edna but with postgres in docker changing password env doesn't has effect.
This is because at first start postgres creates db and roles and do not change password for role on env change. Simply we can remove volume to recreate it.

btw, about your question about russian letters in password - it's working